yellowbrick/classifier/boundaries.py

```diff
@@ -0,0 +1,457 @@
+
+    def draw(self, X, y=None, **kwargs):
+        """
+        Called from the fit method, this method creates a decision boundary
+        plot, and if self.scatter is True, it will scatter plot that draws
+        each instance as a class or target colored point, whose location
+        is determined by the feature data set.
+        """
+        # ensure that if someone is passing in another X such as X_test, that
+        # features will be properly handled
+        X = self._select_feature_columns(X)
+
+        num_colors = len(self.classes_) * 2
+        color_cycle = iter(
+            resolve_colors(color=self.colors, num_colors=num_colors))
+        colors = OrderedDict([(c, next(color_cycle))
+                              for c in self.classes_.keys()])
+
+        self.ax.pcolormesh(
+            self.xx,
+            self.yy,
+            self.Z_shape,
+            alpha=self.pcolormesh_alpha,
+            cmap=ListedColormap(colors.values()))
+
+        # Create a data structure to hold the scatter plot representations
+        to_plot = OrderedDict()
+        for index in self.classes_.values():
+            to_plot[index] = [[], []]
+
+        # Add each row of the data set to to_plot for plotting
+        for i, row in enumerate(X):
+            row_ = np.repeat(np.expand_dims(row, axis=1), 2, axis=1)
+            x_, y_ = row_[0], row_[1]
+            # look up the y class name if given in init
+            if self.class_labels is not None:
+                target = self.class_labels[y[i]]
+            else:
+                target = y[i]
+            index = self.classes_[target]
+            to_plot[index][0].append(x_)
+            to_plot[index][1].append(y_)
+
+        # Add the scatter plots from the to_plot function
+        # TODO: store these plots to add more instances to later
+        # TODO: make this a separate function
+
+        if self.show_scatter:
+            for kls, index in self.classes_.items():
+                self.ax.scatter(
+                    to_plot[index][0],
+                    to_plot[index][1],
+                    marker=next(self.markers),
+                    color=colors[kls],
+                    alpha=self.scatter_alpha,
+                    s=30,
+                    edgecolors='black',
+                    label=str(kls),
+                    **kwargs)
+        else:
+            labels = [
+                Patch(color=colors[kls], label=kls)
+                for kls in self.classes_.keys()
+            ]
+            self.ax.legend(handles=labels)
+
+        self.ax.axis('auto')
+
+    def finalize(self, **kwargs):
+        """
+        Finalize executes any subclass-specific axes finalization steps.
+        The user calls poof and poof calls finalize.
+
+        Parameters
+        ----------
+        kwargs: generic keyword arguments.
+
+        """
+        # Divide out the two features
+        feature_one, feature_two = self.features_
+
+        if self.title is None:
```

Comment:
This is also handled by the base class `set_title` function; you should feel free to just pass your title into set_title.